components/connect_v2/client/index.coffee

```diff
@@ -1,22 +1,30 @@
-module.exports = ($el) ->
-  $input = $el.find 'input'
-  $channels = $el.find 'a'
+ConnectView = require './view.coffee'
 
-  $input
-    .on 'input', (e) ->
-      e.preventDefault()
+{ last } = require 'underscore'
+{ API_URL, CURRENT_USER } = require('sharify').data
+Block = require '../../../models/block.coffee'
+Channels = require '../../../collections/connection_blocks.coffee'
+mediator = require '../../../lib/mediator.coffee'
 
-      $el.attr 'data-state', if $input.val() isnt '' then 'active' else 'inactive'
+module.exports = ($el) ->
+  block = new Block
+  channels = new Channels [], user_slug: CURRENT_USER.slug
 
-      console.log $input.val()
+  replenish = if (recentConnections = mediator.shared.recent_connections)?
+    recentConnections
+      .fetch().then (response) =>
+        channels.add(last(response, 3).reverse())
 
-  $channels
-    .on 'click', (e) ->
-      e.preventDefault()
+  else
+    Promise.resolve()
 
-      $(this).attr 'data-selected'
+  replenish.then ->
+    channels.fetch(data: per: 3 - channels.length) if channels.length < 0
```

Comment:
I suppose it's due to the fact that getting items from localStorage is asynchronous.